services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/extensions/dynamodb/mappingclient/model/GlobalSecondaryIndex.java

```diff
@@ -25,17 +25,25 @@
     private final Projection projection;
     private final ProvisionedThroughput provisionedThroughput;
 
-    private GlobalSecondaryIndex(String indexName, Projection projection, ProvisionedThroughput provisionedThroughput) {
-        this.indexName = indexName;
-        this.projection = projection;
-        this.provisionedThroughput = provisionedThroughput;
+    private GlobalSecondaryIndex(Builder builder) {
+        this.indexName = builder.indexName;
+        this.projection = builder.projection;
+        this.provisionedThroughput = builder.provisionedThroughput;
     }
 
     public static GlobalSecondaryIndex create(String indexName,
                                               Projection projection,
                                               ProvisionedThroughput provisionedThroughput) {
 
-        return new GlobalSecondaryIndex(indexName, projection, provisionedThroughput);
+        return builder().indexName(indexName).projection(projection).provisionedThroughput(provisionedThroughput).build();
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public Builder toBuilder() {
+        return new Builder().indexName(indexName).projection(projection).provisionedThroughput(provisionedThroughput);
```

Comment:
create() calls builder() while toBuilder does new Builder()

Should we make them consistent here and everywhere else?